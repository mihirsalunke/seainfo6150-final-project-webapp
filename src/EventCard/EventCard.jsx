import React from "react";
import styles from "./EventCard.module.css";
import PropTypes from "prop-types";

const EventCard = (props) => {
    return (
        <div className={styles.container}>
            <div className={styles.wrapper}>
                <img src={props.event.image} alt={props.event.title}/>
                <h2 className={styles.title}>{props.event.title}</h2>
                <p className={styles.date}><time dateTime={props.event.date}>{props.event.eventDate}</time></p>
                <p className={styles.location}>Location: <span>{props.event.location}</span></p>
            </div>
        </div>
    );
};

EventCard.propTypes = {
    event: PropTypes.object.isRequired,
};

export default EventCard;