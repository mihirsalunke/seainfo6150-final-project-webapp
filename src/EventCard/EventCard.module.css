@import url("https://fonts.googleapis.com/css2?family=Rowdies:wght@300&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Nerko+One&display=swap");

.container {
    background: white;
    border: 1px solid lightblue;
    border-radius: 10px;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.3);
    padding: 20px;

    display: flex;
    flex-flow: column;
    justify-content: space-between;
    transition: transform .25s ease-out;
}

.container:hover {
    transform: scale(1.25);
}

.container:active {
    transform: scale(1.1);
}

img {
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 100%;
}


.title {
    color: rgb(22, 28, 39);
    font-family: "Nerko One", sans-serif;
    font-size: 2em;
    margin: 0 0 20px;
    text-align: center;
}

.date {
    text-align: center;
    color: orange;
    font-family: Rowdies, sans-serif;
}

.location {
    font-weight: bold;
    text-align: center;
    color: rgb(22, 28, 39);
}

.location span {
    font-family: Rowdies, sans-serif;
}

@media screen and (min-width: 525px) {
    .container {
      padding: 20px;
    }
    .title {
      font-size: 1.5em;
    }
  }
  
  @media screen and (min-width: 800px) {
    .container {
      padding: 15px;
    }
    .title {
      font-size: 1.5em;
    }
  }
  
  @media screen and (min-width: 1150px) {
    .container {
      padding: 10px;
    }
    .title {
      font-size: 2.5em;
    }
  }

