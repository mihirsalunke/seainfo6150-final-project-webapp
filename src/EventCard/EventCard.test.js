import React from "react";
import EventCard from "./EventCard.jsx";
import { render } from "@testing-library/react";

const event = {
    "eventId": "1110001",
    "title": "Future of AI",
    "description": "Future of AI is showcasing latest innovation and business modules",
     "image" : "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS9tSKBQBaTBTW3fMOh_8KPz5CqGeT6AeptUA&usqp=CAU",
      "location": "Chicago",
     "organizerName": "B2C Tech Inc",
      "organizerContactNumber": "351-800-0120",
      "category": "Tech",
     "date": "2020-12-24",
     "time": "13:00",
      "eventDate": "December 24th 2020",
      "eventTime": "1:00 pm"
};

describe("EventCard tests", () => {
    it("renders correctly", () => {
      const { container } = render(<EventCard event={event} />);
      expect(container).toMatchSnapshot();
    });
});