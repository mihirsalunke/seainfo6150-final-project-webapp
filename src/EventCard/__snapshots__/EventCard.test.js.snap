// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="wrapper"
    >
      <img
        alt="Future of AI"
        src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS9tSKBQBaTBTW3fMOh_8KPz5CqGeT6AeptUA&usqp=CAU"
      />
      <h2
        class="title"
      >
        Future of AI
      </h2>
      <p
        class="date"
      >
        <time
          datetime="2020-12-24"
        >
          December 24th 2020
        </time>
      </p>
      <p
        class="location"
      >
        Location: 
        <span>
          Chicago
        </span>
      </p>
    </div>
  </div>
</div>
`;
